crates/turbopack-ecmascript/src/lib.rs

```diff
@@ -353,7 +353,7 @@ impl EcmascriptModuleAsset {
         availability_info: Value<AvailabilityInfo>,
     ) -> Result<Vc<EcmascriptModuleContent>> {
         let this = self.await?;
-        if *self.analyze().needs_availability_info().await? {
+        let availability_info = if *self.analyze().needs_availability_info().await? {
```

Comment:
Uh, what does this bug fix affect? Is this part of the reason for our poor chunking?

Reply:
It only affects caching. It idea was to not pass availability_info when it's not needed to leverage caching between mutliple calls (with different availability_info). The bug only makes caching worse, especially for client components (which have different availability_info each).